app/controllers/api/v0x1/mixins/index_mixin.rb

```diff
@@ -0,0 +1,27 @@
+module Api
+  module V0x1
+    module Mixins
+      module IndexMixin
+        def scoped(relation)
+          if relation.model.respond_to?(:taggable?) && relation.model.taggable?
+            ref_schema = {relation.model.tagging_relation_name => :tag}
+
+            relation = relation.includes(ref_schema).references(ref_schema)
+          end
+          relation
+        end
+
+        def collection(base_query)
+          resp = ManageIQ::API::Common::PaginatedResponse.new(
+            :base_query => base_query,
```

Comment:
Why not `:base_query => scoped(base_query),` and then call `collection(< base_query>)` from `index` methods?

Reply:
Good point, changed.